src/modules/folder/folder.repository.ts

```diff
@@ -447,6 +453,25 @@ export class SequelizeFolderRepository implements FolderRepository {
     return folders.map((folder) => this.toDomain(folder));
   }
 
+  async calculateFolderSize(
```

Comment:
The timeout is here, nice then! I will just use the 15-second timeout inside instead of offering a param for that, as it is a matter of the Infra and in fact, a concern of it (a concern of the repo, and not of some use case - business logic). 

But this code is not aborting the query on the database, is just resolving before, but the query will continue running, thus, the possibility of attacking the database and sending requests to keep it busy is still possible. 

Please abort the query operation when the timeout expires. Also, did we agree on that or rather than it, to return the size calculated until X items / X size?

Reply:
I fixed this using the postgres timeout in the query ( SET LOCAL statement_timeout to :timeoutValue )